LandSandBoat/server
# Environment Variable Refactor + Remove Docker files

**_I affirm:_**

## What does this pull request do?

Update settings.cpp to allow for arbitrary configuration of environment variables using the template XI_{outerKey}_{innerKey}.

Updates to the docker related files to configure everything through the environment, as well as add a reverse proxy to make server available from non-localhost boxes by default.

## Steps to test these changes

Ignoring the Docker stuff, if you set any environment variables before you start like XI_NETWORK_SQL_HOST, or XI_MAIN_SERVER_NAME, it should be automatically reflected in the logs and the runtime behavior when starting up the server.
